File: Java/프로그래머스/3/68646. 풍선 터트리기/풍선 터트리기.java
class Solution {
    public int solution(int[] a) {
        int answer = 0;
        int[] left = new int[a.length];
        int[] right = new int[a.length];
        int first = a[0];
        for (int i = 0; i < a.length; i++) {
            if (a[i] < first) {
                first = a[i];
            }
            left[i] = first;
        }
        int last = a[a.length - 1];
        for (int i = a.length - 1; i >= 0; i--) {
            if (a[i] < last) {
                last = a[i];
            }
            right[i] = last;
        }

        for (int i = 0; i < a.length; i++) {
            if (!(left[i] < a[i] && right[i] < a[i])) {
                answer++;
            }
        }
        return answer;
    }
}

File: Python/프로그래머스/3/68646. 풍선 터트리기/풍선 터트리기.py
def solution(a):
    answer = 0
    left = [0] * len(a)
    right = [0] * len(a)

    first = a[0]
    for i in range(len(a)):
        if a[i] < first:
            first = a[i]
        left[i] = first

    last = a[len(a) - 1]
    for i in range(len(a) - 1, -1, -1):
        if a[i] < last:
            last = a[i]
        right[i] = last

    for i in range(len(a)):
        if not (left[i] < a[i] and right[i] < a[i]):
            answer += 1

    return answer